src/common/utils/oidc/helper.go

```diff
@@ -119,12 +121,13 @@ type Token struct {
 // UserInfo wraps the information that is extracted via token.  It will be transformed to data object that is persisted
 // in the DB
 type UserInfo struct {
-	Issuer        string   `json:"iss"`
-	Subject       string   `json:"sub"`
-	Username      string   `json:"name"`
-	Email         string   `json:"email"`
-	Groups        []string `json:"groups"`
-	hasGroupClaim bool
+	Issuer           string   `json:"iss"`
+	Subject          string   `json:"sub"`
+	Username         string   `json:"name"`
+	Email            string   `json:"email"`
+	Groups           []string `json:"groups"`
+	AdminGroupMember bool     `json:"AdminGroupMember"`
```

Comment:
The AdminGroupMember should not put in the UserInfo, because when AdminGroup setting is changed, then previous admin user should lost the admin privilege at once, when the AdminGroupMember is persitent in token and saved in database, the user still have the admin privilege when the token is valid. because the groups is already contains the group information, we can use info to check it has admin role.

Reply:
Thanks for the review, but user info is tied to the id token, so each time Harbor gets a new id token it will extracts the admin group from setting and compare with token and populate to user info, then to user model.

It's like removing admin role from a user, this will take effect after user re-authenticate.

Moving the code to `mergeUserInfo` makes no improvement.

Reply:
And for clarification the adminGroup flag is not in the token or persisted. 
Each time the token is verified the flag will be regenerated based on the OIDC settings in Harbor.